app/helpers/hosts_helper.rb

```diff
@@ -253,50 +238,13 @@ def build_duration(host)
-    fields += [[_("Build errors"), link_to("Logs from OS installer", build_errors_host_path(:id => host.id))]] if host.build_errors.present?
-    fields += [[_("Token"), host.token || _("N/A")]] if User.current.admin?
-    fields += [[_("Domain"), link_to(host.domain, hosts_path(:search => %{domain = "#{host.domain}"}))]] if host.domain.present?
-    fields += [[_("Realm"), link_to(host.realm, hosts_path(:search => %{realm = "#{host.realm}"}))]] if host.realm.present?
-    fields += [[_("IP Address"), host.ip]] if host.ip.present?
-    fields += [[_("IPv6 Address"), host.ip6]] if host.ip6.present?
-    fields += [[_("Comment"), host.comment]] if host.comment.present?
-    fields += [[_("MAC Address"), host.mac]] if host.mac.present?
-    fields += [[_("Puppet Environment"), link_to(host.environment, hosts_path(:search => %{environment = "#{host.environment}"}))]] if host.environment.present?
-    fields += [[_("Architecture"), link_to(host.arch, hosts_path(:search => %{architecture = "#{host.arch}"}))]] if host.arch.present?
-    fields += [[_("Operating System"), link_to(host.operatingsystem.to_label, hosts_path(:search => %{os_title = "#{host.operatingsystem.title}"}))]] if host.operatingsystem.present?
-    fields += [[_("PXE Loader"), host.pxe_loader]] if host.operatingsystem.present? && !host.image_build?
-    fields += [[_("Host group"), link_to(host.hostgroup, hosts_path(:search => %{hostgroup_title = "#{host.hostgroup}"}))]] if host.hostgroup.present?
-    fields += [[_("Uptime"), time_ago_in_words(host.uptime_seconds.seconds.from_now)]] if host.uptime_seconds.present?
-    fields += [[_("Location"), (link_to(host.location.title, hosts_path(:search => %{location = "#{host.location}"})) if host.location)]] if SETTINGS[:locations_enabled]
-    fields += [[_("Organization"), (link_to(host.organization.title, hosts_path(:search => %{organization = "#{host.organization}"})) if host.organization)]] if SETTINGS[:organizations_enabled]
-    if host.owner_type == _("User")
-      fields += [[_("Owner"), (link_to(host.owner, hosts_path(:search => %{user.login = "#{host.owner.login}"})) if host.owner)]]
-    else
-      fields += [[_("Owner"), host.owner]]
+    fields = []
+    UI::HostDescription.reduce_providers(:overview_fields).each do |provider|
+      fields += send(provider, host)
     end
-    fields += [[_("Certificate Name"), host.certname]] if Setting[:use_uuid_for_certificates]
-    fields
-  end
-
-  def host_detailed_status_list(host)
-    host.host_statuses.sort_by(&:type).map do |status|
-      next unless status.relevant? && !status.substatus?
-      [
-        _(status.name),
-        content_tag(:span, ' '.html_safe, :class => host_global_status_icon_class(status.to_global)) +
-          content_tag(:span, _(status.to_label), :class => host_global_status_class(status.to_global))
-      ]
-    end.compact
+    priority_buttons.
+      sort_by { |field_hash| field_hash[:priority] }.
+      map { |field_hash| field_hash[:field] }
```

Comment:
Since other helpers use the same sorting logic, maybe we could have something like:

```ruby
def overview_fields(host)
 #...
  sort_fields fields, :field
end

def sort_fields(fields, field_key)
  fields.sort_by { |field_hash| field_hash[:priority] }.map { |field_hash| field_hash[field_key] }
end

```